### PR TITLE
Apply ktlinter to the project

### LINKED ISSUE #442: Add klint for coding style enforcement
# Feature Request

## Description
Currently there is no coding style enforcement whatsoever but now that there is an [official standard]( https://kotlinlang.org/docs/reference/coding-conventions.html) we should try to adhere to it. This can be enforced and easily checked by adding a linter to both the codebase as well as the build process.

## Proposed Solution
- [ ] Add https://github.com/shyiko/ktlint by adding it to the Gradle configuration
- [ ] Add documentation to show how to enable it in Intellij (and thus Android Studio)
- [ ] Add klint to travis

### PR DESCRIPTION
**Pull request by kittinunf**
## Description

In order to avoid counterproductive comments like "You forget to add space here" or "There is no newline there", this PR introduces Ktlint to help take care lint in the project. This PR will introduce [kotlinter-gradle](https://github.com/jeremymailen/kotlinter-gradle) as it is super duper easy to integrate. Merging this closes #442.

## Type of change

Check all that apply

- [x] Add helper for Lint

## How Has This Been Tested?

Travis should scream at us if we violate the rules.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if necessary
- [ ] My changes generate no new compiler warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -14,6 +14,7 @@ plugins {
     id(Jacoco.Android.plugin) version Jacoco.Android.version apply false
     id(BintrayRelease.plugin) version BintrayRelease.version apply false
     id(KotlinX.Serialization.plugin) version Kotlin.version apply false
+    id(Ktlint.plugin) version Ktlint.version apply false
```
Comment (kittinunf) [inline build.gradle.kts]: The "apply false" thingy is necessary? @NikkyAI

**Comment by kittinunf** [comment]
Okay it looks like it works!

``` 
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:29:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:30:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:41:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:52:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:63:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:74:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:85:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:96:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:107:9: Missing space after //
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:155:33: Unexpected spacing before ")"
Lint error > src/main/kotlin/com/github/[secure]/fuel/Fuel.kt:157:45: Unexpected spacing before ")"
```

**Comment by markGilchrist** [comment]
@kittinunf  looks good to me, I am going to leave this to @NikkyAI to approve as (s)he has deeper insight into Kotlin gradle files than I do

File: build.gradle.kts (hunk 1 above)
Comment (kittinunf) [reply build.gradle.kts]: Let's delete it then? I feel it is annoying

**Comment by kittinunf** [comment]
Okay I am gonna 💣 💥 this PR with fixes.

File: build.gradle.kts (hunk 1 above)
Comment (kittinunf) [reply build.gradle.kts]: Oh no ... it looks like it is necessary 😢

**Comment by NikkyAI** [comment]
Can we add a guide to the rREADME how to set up idea with the correct stle rules or at least link to the setup instructions on the ktlint readme ?
and/or link it in CONTRIBUTING.md

**Comment by NikkyAI** [comment]
might be nice to have `./gradlew formatKotlin` as a commit hook and also instructions for setup

**Comment by kittinunf** [comment]
@NikkyAI Thanks! Please review one more time 🙏

**Comment by NikkyAI** [comment]
looks good to merge to me

**Pull request merged**